src/controller/CHIPDeviceController.cpp

```diff
@@ -3207,6 +3233,52 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
         }
     }
     break;
+    case CommissioningStage::kGetTCAcknowledgments: {
+        ChipLogProgress(Controller, "Get Terms and Conditions Acknowledgments");
+
+        // If terms and conditions acknowledgements are not required, or have already been provided, then proceed
+        if (!params.GetRequireTermsAndConditionsAcknowledgement() || params.GetTermsAndConditionsAcknowledgement().HasValue())
+        {
+            TermsAndConditionsAcknowledgementsReady();
+            return;
+        }
+
+        ChipLogProgress(Controller, "Waiting for Terms and Conditions");
```

Comment:
Doesn't this need to extend the fail-safe or something?  What happens if the user takes 10 minutes to read the terms (which is a bare minimum for how long it can possibly take to actually read those things typically)?

Reply:
We talked offline. I removed the wait-state. The user input related to T&C acknowledgements availability now needs to be a precondition to starting AutoCommissioner.